### DISCUSSION TITLE
플레이어에 붙여준 커스텀스크립트에서 다른 오브젝트 스크립트 GetComponent시 캐릭터 생성 안되는 문제

### DISCUSSION THREAD

**Discussion by LeeYongSu1** [naverz/zepeto-studio-kor#176] [5. Q&A]
OnAddedLocalPlayer에 커스텀스크립트를 붙이는 부분은 정상 작동이 되고 있습니다. 다만, 붙인 커스텀스크립트에서 특정 오브젝트가 가지고 있는 스크립트를 GetComponent하는 구문이 들어가면 정상 작동이 되지 않습니다.

OnAddedLocalPlayer 타이밍에 커스텀스크립트Add -> _붙인 커스텀스크립트 runtime 타이밍에 오브젝트 찾아서 GetComponet<다른 커스텀스크립트 or 빈 스크립트>()를 하는 과정_입니다.

이 과정(붙인 커스텀스크립트 runtime 타이밍)에 자기 자신의 컴포넌트를 가져오거나, 다른 오브젝트에 접근해서 gameObject나 Transform을 가져오는 부분을 잘 동작을 하는데 스크립트를 GetComponent하면 동작을 하지 않습니다.

<예제 ClientStarter에서 커스텀 스크립트 Add>
![image](https://user-images.githubusercontent.com/80555875/148476839-731af946-6ad8-4555-b6b3-87dc56bf8a10.png)

<Add된 커스텀스크립트에서 다른 오브젝트 접근 후 스크립트 Get>
![image](https://user-images.githubusercontent.com/80555875/148476861-6c548784-1248-43ed-bd9d-5a8bbe8f441d.png)

<Get하는 스크립트>
![image](https://user-images.githubusercontent.com/80555875/148476873-2c3e4ae6-b6c3-43bb-8020-e2bd272a70a9.png)

동적으로 붙여준 스크립트에서 다른 스크립트에 접근하려고 하면 문제가 발생한다고 생각되는데 확인 부탁드려도 될까요?

**Comment by moondory77-naverz** [comment]
안녕하세요.
올려주신 예제를 기준으로 테스트 할 때
ZEPETO.World 1.1.2 기준, 
커스텀 스크립트에서 커스텀 스크립트로 접근에 문제는 발견되지 않았습니다.

다른 부분에 문제가 있을 지, 확인을 부탁드리겠습니다.
감사합니다.

**Reply by LeeYongSu1** [reply]
넵 패키지 업데이트 후 다시 확인해보도록 하겠습니다
